src/Sentry/Integrations/WinUIUnhandledExceptionIntegration.cs

```diff
@@ -0,0 +1,133 @@
+    // For UWP, the calling application will need to hook the event handler.
+    // See https://docs.sentry.io/platforms/dotnet/guides/uwp/
+    // (We can't do it automatically without a separate UWP class library,
+    // due to a security exception when attempting to attach the event dynamically.)
+
+    internal class WinUIUnhandledExceptionIntegration : ISdkIntegration
+    {
+        private static readonly Assembly? WinUIAssembly = GetWinUIAssembly();
+
+        private Exception? _lastFirstChanceException;
+        private IHub _hub = null!;
+        private SentryOptions _options = null!;
+
+        // Constants used for reflection
+        private const string WinUIAssemblyName = "Microsoft.WinUI";
+        private const string WinUiAssemblyPublicKeyToken = "de31ebe4ad15742b";
+        private const string WinUINamespace = "Microsoft.UI.Xaml";
+        private const string ApplicationTypeName = "Application";
+        private const string UnhandledExceptionEventName = "UnhandledException";
+        private const string CurrentPropertyName = "Current";
+        private const string HandledPropertyName = "Handled";
+        private const string ExceptionPropertyName = "Exception";
+        private const string ApplicationTypeFullName = WinUINamespace + "." + ApplicationTypeName;
+        private const string Mechanism = WinUINamespace + "." + UnhandledExceptionEventName;
+
+        public static bool IsApplicable => WinUIAssembly != null;
+
+        public void Register(IHub hub, SentryOptions options)
+        {
+            if (!IsApplicable)
+            {
+                return;
+            }
+
+            _hub = hub;
+            _options = options;
+
+            // Hook the main event handler
+            AttachEventHandler();
+
+            // First part of workaround for https://github.com/microsoft/microsoft-ui-xaml/issues/7160
+            AppDomain.CurrentDomain.FirstChanceException += (_, e) => _lastFirstChanceException = e.Exception;
+        }
+
+        private static Assembly? GetWinUIAssembly()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // Not running on Windows
+                return null;
+            }
+
+            // Attempt to locate the Microsoft.WinUI assembly from the AppDomain
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies();
+            return Array.Find(assemblies, x =>
+            {
+                // first check by name
+                var assemblyName = x.GetName();
+                if (assemblyName.Name != WinUIAssemblyName)
+                {
+                    return false;
+                }
+
+                // check the public key token also
+                var token = assemblyName.GetPublicKeyToken();
+                return token != null && string.Equals(Convert.ToHexString(token), WinUiAssemblyPublicKeyToken, StringComparison.OrdinalIgnoreCase);
```

Comment:
would be better to convert `WinUiAssemblyPublicKeyToken` to bytes once at startup. then to a byte array compare